src/net/tcp/listener.rs

```diff
@@ -31,9 +31,9 @@ use std::{io, net::SocketAddr};
 ///
 ///     tx.write(b"test" as &'static [u8]).await.0.unwrap();
 ///
-///     let (_, buf) = rx.read(vec![0; 4]).await;
+///     let (_, slice) = rx.read(vec![0; 4]).await;
 ///
-///     assert_eq!(buf, b"test");
+///     assert_eq!(slice.to_vec(), b"test");
```

Comment:
That's cool. Almost magic that the Slice has to_vec without this crate defining it. That's what ops::Deref gives us?

Reply:
Yes that's available through `Deref`. But it gives a bad usage example, there is no need to copy the data. Fixed this to access the byte slice through `Deref`.